webserver.py

```diff
@@ -41,61 +43,205 @@ def get_conn():
+        game_state = GameState.load(slug, password)
+        if game_state:
+            raise Forbidden("Username already taken")
+        else:
+            game_state = GameState({
+                'resources': {},
+                'slug': slug,
+                'password': password,
+                'seedCounts': {GAME_CONFIG['firstSeed']: 0},
+                'recipes': [GAME_CONFIG['firstSeed']],
+                'plots': {}
+            })
+        return game_state.initialize()
+
+    @classmethod
+    def load(cls, slug, password):
+        my_conn = get_conn()
+        with my_conn:
+            with my_conn.cursor() as cursor:
+                cursor.execute('SELECT game_state FROM game_states WHERE slug = %s;', [slug])
+                row = cursor.fetchone()
+                if not row:
+                    return None
+                game_state = GameState(row[0])
+                game_state.check_password(password)
+                return game_state.initialize()
+
+    def save(self, slug):
+        my_conn = get_conn()
+        json_game_state = json.dumps(self.data, indent=4)
+        with my_conn:
+            with my_conn.cursor() as cursor:
+                cursor.execute('INSERT INTO game_states (slug, game_state) VALUES (%s, %s) '
+                               'ON CONFLICT (slug) DO UPDATE SET game_state=%s;',
+                               [slug, json_game_state, json_game_state])
+
+    def initialize(self):
+        for resource in GAME_CONFIG['starting_resources']:
+            if self.data['resources'].get(resource) is None:
+                if GAME_CONFIG['starting_resources'][resource] > 0:
+                    self.data['resources'][resource] = GAME_CONFIG['starting_resources'][resource]
+        if self.data.get('plots') is None:  # temp
+            self.data['plots'] = {}
+        if 'unlockCount' in self.data:      # temp
+            del self.data['unlockCount']
+        for i in range(GAME_CONFIG['field_width']):
+            for j in range(GAME_CONFIG['field_height']):
+                if 'plot' + str(i) + '_' + str(j) in self.data:
+                    if self.data['plot' + str(i) + '_' + str(j)]['locked'] == 0:
+                        self.data['plots'][str(i) + '_' + str(j)]['seedType'] = self.data['plot' + str(i) + '_' + str(j)]['seedType']
+                        self.data['plots'][str(i) + '_' + str(j)]['sowTime'] = self.data['plot' + str(i) + '_' + str(j)]['sowTime']
+                    del self.data['plot' + str(i) + '_' + str(j)]
+        for i in range(2):
+            for j in range(2):
+                if self.data['plots'].get(str(i) + '_' + str(j)) is None:
+                    self.data['plots'][str(i) + '_' + str(j)] = {'seedType': 0, 'sowTime': 0}
+        return self
+
+    def check_cash(self):
+        total_seed_count = 0
+        total_crops_planted = 0
+        for seed_id in GAME_CONFIG['seeds']:
+            total_seed_count += self.data['seedCounts'][seed_id]
```

Comment:
This spot's going to give you trouble-- if you intialized `data['seedCounts']` with just {'a': 0}, when it counts through all the seed_ids available it's going to fail a bunch of times.

A simple fix is to use `.get()` or create a `get_seed_count(seed_id)` method on GameState

A slightly more efficient way would be to just iterate over the seeds you do have in `data['seedCounts']` because you don't care about the other seed_ids-- they're going to be 0.

```
for seed_id in self.data['seedCounts']:
    total_seed_count += self.data['seedCounts'][seed_id]
```